src/Extensions/Features/src/FeatureCollectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http.Features;
+
+/// <summary>
+/// Extension methods for getting feature from <see cref="IFeatureCollection"/>
+/// </summary>
+public static class FeatureCollectionExtensions
+{
+    /// <summary>
+    /// Retrives the requested feature from the collection.
+    /// Throws an <see cref="InvalidOperationException"/> if the feature is not present.
+    /// </summary>
+    /// <param name="featureCollection">The <see cref="IFeatureCollection"/>.</param>
+    /// <typeparam name="TFeature">The feature key.</typeparam>
+    /// <returns>The requested feature.</returns>
+    public static TFeature GetRequiredFeature<TFeature>(this IFeatureCollection featureCollection)
+        where TFeature : notnull
+    {
+        if (featureCollection is null)
+        {
+            throw new ArgumentNullException(nameof(featureCollection));
+        }
+
+        return featureCollection.Get<TFeature>() ?? throw new InvalidOperationException($"Feature {typeof(TFeature)} is not present.");
```

Comment:
```suggestion
        return featureCollection.Get<TFeature>() ?? throw new InvalidOperationException($"Feature '{typeof(TFeature)}' is not present.");
```

Reply:
Why single quotes?

Reply:
It's something we almost always do when we include parameterized type names in exception messages to make it more obvious that it's a generic error that could happen with different types. We do it all over DI including `GetRequiredService`:

https://github.com/dotnet/runtime/blob/64f48857a6193a81faa32825cc78d7340a3ebd46/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/Resources/Strings.resx#L137

We also single quoted type names in some of the error messages this PR is replacing:

https://github.com/dotnet/aspnetcore/pull/39192/files#diff-abeea620ed1d2f72f8243f6600f3dcae70403d3186190f5131b4f909b9173b2fL371

Reply:
@halter73 could we get this PR merged and do a clean up pass?

Reply:
There are still failing tests to address, so cleanup is harmless.